Support `cpu.frequency` hardware requirement for ``mrack``

skycastlelily opened the pull request:
Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] extend the test coverage
* [x] update the specification
* [x] mention the version
* [x] include a release note

happz commented:
Blocked on https://github.com/teemtee/tmt/pull/3295

happz commented on spec/hardware/cpu.fmf:
```diff
@@ -70,7 +70,7 @@ description: |
 
     .. versionchanged:: 1.38
-       ``beaker`` plugins supports ``stepping``
+       ``beaker`` plugins supports ``frequency`` and ``stepping``
```
Both release note and this `versionchanged` need an update, it's no longer 1.38 but 1.39.

thrix replied:
addressed

skycastlelily commented:
Rebased and updated,and it just occurred to me that I should update first
in order to put the code addressing the comments in one commit, then
rebased, will keep that in mind^^

On Sat, Nov 2, 2024 at 5:20 PM Miloš Prchlík ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In spec/hardware/cpu.fmf
> <https://github.com/teemtee/tmt/pull/3297#discussion_r1826532800>:
>
> > @@ -70,7 +70,7 @@ description: |
>      are also useful resources.
>
>      .. versionchanged:: 1.38
> -       ``beaker`` plugins supports ``stepping``
> +       ``beaker`` plugins supports ``frequency`` and ``stepping``
>
> Both release note and this versionchanged need an update, it's no longer
> 1.38 but 1.39.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/teemtee/tmt/pull/3297#pullrequestreview-2411221343>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKFR23FEK5YSOZBKUJIKMC3Z6SKOXAVCNFSM6AAAAABQDF2DPOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMJRGIZDCMZUGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

skycastlelily commented:
Rebased and updated,and it just occurred to me that I should update first in order to put the code addressing the comments in one commit, then rebased, will keep that in mind^^

martinhoyer commented on spec/hardware/cpu.fmf:
```diff
@@ -115,15 +118,20 @@ example:
         hyper-threading: true
 
+  - |
+    # Request a CPU with specified frequency.
```
There is already the frequency example above, stating MHz is default and that it accepts different units, maybe best to not add another example?
https://tmt.readthedocs.io/en/stable/spec/hardware.html#cpu

skycastlelily replied:
Updated,thanks for your review:)

martinhoyer commented on tmt/hardware.py:
```diff
@@ -737,7 +737,7 @@ def _cast_int(raw_value: Any) -> int:
 
 
-class NumberConstraint(Constraint[float]):
+class NumberConstraint(Constraint['Size']):
```
hmm, I might be missing something. Why this change?

happz replied:
Doesn't look correct indeed.

happz replied:
Same here ;)

skycastlelily replied:
update:https://github.com/teemtee/tmt/pull/3297/commits/7df0e709e1addbd4ebc62955511e43de37fd341a.

Sorry,mentor, I somehow overlooked the later part of the sentence :sweat_smile:

therazix commented on spec/hardware/cpu.fmf:
```diff
@@ -125,8 +125,8 @@ example:
+    note: "``cpu.vendor``, ``cpu.vendor-name``, ``cpu.frequency`` and ``cpu.hyper-threading`` not implemented yet"
   - implemented-by: /tmt/steps/provision/mrack.py
-    note: "``cpu.flag``, ``cpu.processors``, ``cpu.model``, ``cpu.cores``, ``cpu.model-name``, ``cpu.hyper-threading``, ``cpu.stepping``, ``cpu.family`` and ``cpu.vendor-name`` only"
+    note: "``cpu.sockets``, ``cpu.threads``, ``cpu.cores-per-socket``, ``cpu.threads-per-core`` not implemented yet"
```
Shouldn't `family-name` and `vendor` also be mentioned here?
```suggestion
    note: "``cpu.sockets``, ``cpu.threads``, ``cpu.cores-per-socket``, ``cpu.threads-per-core``, ``cpu.family-name``, ``cpu.vendor`` not implemented yet"
```

skycastlelily replied:
Update:https://github.com/teemtee/tmt/pull/3297/commits/2a293b53b86df42771ce5bc6eaffdf1a8f356bb4
Thanks for your review:)

psss reviewed (approved):
Looks good, thanks!

psss commented:
Red jobs are irrelevant failures, merged.

skycastlelily commented:
:tada:
